### PR TITLE
scripts: introduce script to apply email, working around google groups brokeness

### PR DESCRIPTION
**Pull request by avikivity**
Google Groups recently started rewriting the From: header, garbaging
our git log. This script rewrites it back, using the Reply-To header
as a still working source.

### CODE REVIEW

**Comment by avikivity** [comment]
@scylladb/scylla-maint please use once merged

File: scripts/apply-mail.py
Comment (denesb) [inline scripts/apply-mail.py:21]: I don't think `message_from_file` takes ownership of the file (doesn't close it). The python way of doing this:
```python
with open(email_file) as f:
    e = email.message_from_file(f)
```
You don't have to worry about block scope here (`e` not being available outside `with` block), python doesn't have block scope.

**Review by denesb — APPROVED** [review]
LGTM
One minor nit (not closing a file is inconsequential for such a tiny script, it is closed at teardown anyway).

**Comment by denesb** [comment]
> @scylladb/scylla-maint please use once merged

You can add a mention to maintainer's handbook about it.

File: scripts/apply-mail.py
Comment (avikivity) [reply scripts/apply-mail.py:21]: It will auto-close when it falls out of scope. message_for_file() will take a reference on the opened file, and as soon as it exits, the file will be closed.

**Comment by avikivity** [comment]
v2:
 - updated directions in maintainer's handbook

File: scripts/apply-mail.py
Comment (nyh) [inline scripts/apply-mail.py:26]: An idea (which sounds like a func exercise but isn't critical):

The risk of the NamedTemporaryFile() function is that interrupting this code can leave it behind. Traditional Unix code traps interrupts in this case to delete the temporary file, but you don't do it. TemporaryFile is safer in this regard - the file is immediately deleted.
But how can we use the unnamed temporary file in "git am"? You can, using /proc/self/fd/...

**Review by nyh — APPROVED** [review]

File: scripts/apply-mail.py
Comment (avikivity) [reply scripts/apply-mail.py:26]: Python will delete the file if interrupted, unless by a SIGKILL. I think the damage from a few text files in /tmp is low enough to ignore it.

Comment (nyh) [reply scripts/apply-mail.py:26]: @avikivity it turns out you're right **theoretically**, because Python does catch SIGINT and turn it into an exception, and exceptions are handled correctly. It turns out that in practice, the implementation isn't perfect (https://github.com/python/cpython/issues/78338), but you're right that for what we're doing (a script for developers), it's good enough.

File: scripts/apply-mail.py
Comment (xemul) [inline scripts/apply-mail.py:32]: Maybe it's better to stop before calling `git am`? So the script is then called `fix-mail-from.py` or something.
I don't always use `-i` key, and sometimes (but maybe it was some other project) I had to use the `--no-scissors` one, so this hard-coded git-am CLI will cause some inconvenience.

Comment (nyh) [reply scripts/apply-mail.py:32]: I think it's good to have the script do the "git am" otherwise I will forget I even need to use it (if I always use "git am" manually, it's easy to forget there's another step before it).

I agree that the "-i" option isn't necessary and I never use it - simply because you can always follow this with "git commit --amend" if you want to edit the commit message. On the other hand, it isn't so hard to exit the editor too, so I think it's fine that "-i" is the default.

Regarding adding **more** options - we could just add all the extra options that we didn't understand to the git am line. I'm not saying this will be easy with the python parseargs library, though :-( Maybe it's easiest just to add specific options (like the "--no-scissors" one you wanted) and copy them to the command line.

Comment (xemul) [reply scripts/apply-mail.py:32]: > On the other hand, it isn't so hard to exit the editor too...

1_AD1e170YTJaiUBypv1H9Ow

**Pull request closed without merging**
